modules/packages/arch/metadata.go

```diff
@@ -0,0 +1,387 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package arch
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	"code.gitea.io/gitea/modules/packages"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/modules/validation"
+
+	"github.com/mholt/archiver/v3"
+)
+
+// Arch Linux Packages
+// https://man.archlinux.org/man/PKGBUILD.5
+
+const (
+	PropertyDescription  = "arch.description"
+	PropertyArch         = "arch.architecture"
+	PropertyDistribution = "arch.distribution"
+
+	SettingKeyPrivate = "arch.key.private"
+	SettingKeyPublic  = "arch.key.public"
+
+	RepositoryPackage = "_arch"
+	RepositoryVersion = "_repository"
+)
+
+type GlobalVarType struct {
+	reName   *regexp.Regexp
+	reVer    *regexp.Regexp
+	reOptDep *regexp.Regexp
+	rePkgVer *regexp.Regexp
+
+	maxMagicLength int
+	magics         map[string]struct {
+		magic    []byte
+		archiver func() archiver.Reader
+	}
+
+	ArchPkgOrSig *regexp.Regexp
+	ArchDBOrSig  *regexp.Regexp
+}
+
+var GlobalVar = sync.OnceValue[*GlobalVarType](func() *GlobalVarType {
+	v := GlobalVarType{
+		reName:       regexp.MustCompile(`^[a-zA-Z0-9@._+-]+$`),
+		reVer:        regexp.MustCompile(`^[a-zA-Z0-9:_.+]+-+[0-9]+$`),
+		reOptDep:     regexp.MustCompile(`^[a-zA-Z0-9@._+-]+([<>]?=?([0-9]+:)?[a-zA-Z0-9@._+-]+)?(:.*)?$`),
+		rePkgVer:     regexp.MustCompile(`^[a-zA-Z0-9@._+-]+([<>]?=?([0-9]+:)?[a-zA-Z0-9@._+-]+)?$`),
+		ArchPkgOrSig: regexp.MustCompile(`^.*\.pkg\.tar\.\w+(\.sig)*$`),
+		ArchDBOrSig:  regexp.MustCompile(`^.*.db(\.tar\.gz)*(\.sig)*$`),
+	}
+	v.magics = map[string]struct {
+		magic    []byte
+		archiver func() archiver.Reader
+	}{
+		"zst": {
+			magic: []byte{0x28, 0xB5, 0x2F, 0xFD},
+			archiver: func() archiver.Reader {
+				return archiver.NewTarZstd()
+			},
+		},
+		"xz": {
+			magic: []byte{0xFD, 0x37, 0x7A, 0x58, 0x5A},
+			archiver: func() archiver.Reader {
+				return archiver.NewTarXz()
+			},
+		},
+		"gz": {
+			magic: []byte{0x1F, 0x8B},
+			archiver: func() archiver.Reader {
+				return archiver.NewTarGz()
+			},
+		},
```

Comment:
Are there any other containers than zst in the wild? All repositories I have tested just contain zst.